gemd/entity/has_dependencies.py

```diff
@@ -9,5 +9,8 @@ class HasDependencies(ABC):
     """Mix-in trait for objects that reference other objects."""
 
     @abstractmethod
-    def _local_dependencies(self) -> Set[Union["BaseEntity", "LinkByUID"]]:  # noqa: F821
+    def _local_dependencies(
+            self
+    ) -> Set[Union["gemd.entity.base_entity.BaseEntity",  # noqa: F821
+                   "gemd.entity.link_by_uid.LinkByUID"]]:  # noqa: F821
```

Comment:
Nitpick: Why does this list subtypes that can be returned? As the parent class, it doesn't know. It only knows (insofar as `typing` can) that each element will be a subclass of `HasDependencies`.

Reply:
It's not a subclass, it's a mixin / interface specifically designed to say what objects need to be included in a serialization of this object.  It was added to support the register_all method for GEMD objects creating batches of self-supporting objects

https://github.com/CitrineInformatics/citrine-python/blob/54b9fb8a0658dfe0c98315260d1a7e1f972d2a53/src/citrine/_utils/batcher.py#L58

Might make sense to make it an abstract method on BaseEntity instead, except it's used by BaseObject & BaseTemplate & BaseAttribute objects, but not by other Entities.  The object model is really not clean in its dependencies.

Reply:
A valid return value from _local_dependencies is a PropertyTemplate, because something can depend on a PropertyTemplate.  A PropertyTemplate is not a HasDependencies, because it can't depend on anything.  

The return values from _local_dependencies are not necessarily HasDependencies objects.